src/host/ut_host/TextBufferIteratorTests.cpp

```diff
@@ -22,6 +22,61 @@ using namespace WEX::Logging;
 using namespace WEX::TestExecution;
 using Microsoft::Console::Interactivity::ServiceLocator;
 
+template<typename T>
+T GetIterator()
+{
+}

```

Comment:
Two phase lookup goes brrrrrrr...
It goess brrr with our code ordering.

Just like regular functions, templates also need to be defined before they're used in C++. It's just MSVC which was treating templates as fancy macros for the longest time and didn't do that. A lot of our code depends on this behavior.

Reply:
_ouch_. okay this makes sense.

We could also leave them without bodies, correct?

```
template<typename T>
T GetIterator();
```

and then provide specializations? Therefore, if we call it with a T that has not been specialized we get a linker error?

Or does this provide a compiler error?

Actually, how does this  *not* provide an error in a compliant parser? There's no return, it should fail before we get to template generation.

Reply:
Oh, this is code you moved. oops.